lib/pbench/server/cache_manager.py

```diff
@@ -226,6 +238,30 @@ def create(cls, tarball: Path, controller: "Controller") -> "Tarball":
 
         return cls(destination, controller)
 
+    def cache_map(self, tarball: Path):
+        """Builds Hierarchy structure of a Tarball in a Dictionary
+        Format.
+
+        Args:
+            tarball: path of an unpacked tarball
+        """
+        cmap = {tarball.name: {"details": TarInfo(tarball)}}
+        dir_queue = [(tarball, cmap)]
+        while dir_queue:
+            tar_path, parent_map = dir_queue.pop(0)
+            tar_n = tar_path.name
+            curr = parent_map[tar_n]
+
+            dir_list = tar_path.glob("*")
+            for l_path in dir_list:
+                tar_info = TarInfo(l_path)
+                if l_path.is_dir():
+                    curr[l_path.name] = {"details": tar_info}
+                    dir_queue.append((l_path, curr))
+                elif l_path.is_file:
```

Comment:
😆  You're missing the `()` on line 261...so I don't think that condition is checking what we intend for it to check!  (Funny the things that I do and don't notice during a review....)